File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/abstract_factory/AccessDepartment.java
package jv.com.tencent.cain.design_pattern.abstract_factory;

/**
 * @author : jiangyu
 * @date : 2020/3/20
 * @desc : xxx
 */
public class AccessDepartment implements IDepartment {
    @Override
    public void insertDepartment(Department department) {
        System.out.println("access# insert department: " + department);
    }

    @Override
    public Department queryDepartment(int id) {
        return new Department("IOS", 10);
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/abstract_factory/AccessDepartment.kt
package kt.com.tencent.cain.desigin_pattern.abstract_factory

/**
 * @author : jiangyu
 * @date   : 2020/3/20
 * @desc   : xxx
 */
class AccessDepartment : IDepartment {
    override fun insertDepartment(department: Department) {
        println("access# insert department: $department")
    }

    override fun queryDepartment(id: Int): Department = Department("Android", 9)
}